src/main/java/com/app/bookstore/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,128 @@
+        Order order = orderRepository.findOrderByOrderId(orderId);
+        order.setStatus(request.getStatus());
+        return orderMapper.toDto(orderRepository.save(order));
+    }
+
+    @Override
+    public OrderItemDto getItemById(Long orderId, Long itemId) {
+        User user = userService.getAuthenticatedUser();
+        Order order = orderRepository.findOrderByOrderId(orderId);
+        if (Objects.equals(order.getUser().getId(), user.getId())) {
+            OrderItem orderItem = order.getOrderItems().stream()
+                    .filter(item -> item.getId().equals(itemId))
+                    .findFirst()
+                    .orElseThrow(
+                            () -> new EntityNotFoundException("Can't find order item with id "
+                                    + itemId));
+            return orderItemMapper.toDto(orderItem);
+        }
+        throw new EntityNotFoundException("Can't find order with id: " + orderId);
+    }
+
+    @Override
+    public Set<OrderItemDto> getAllOrderItems(Long orderId) {
+        User user = userService.getAuthenticatedUser();
+        Order order = orderRepository.findOrderByOrderId(orderId);
+        if (Objects.equals(order.getUser().getId(), user.getId())) {
+            Set<OrderItem> orderItems = order.getOrderItems();
+            return orderItems.stream()
+                    .map(orderItemMapper::toDto)
+                    .collect(Collectors.toSet());
+        }
+        throw new EntityNotFoundException("Can't find order with id: " + orderId);
+    }
+
+    @Override
+    public Set<OrderDto> getAllOrders(Pageable pageable, Long userId) {
+        return orderRepository.findAllByUserId(pageable, userId)
+                .stream()
+                .map(orderMapper::toDto)
+                .collect(Collectors.toSet());
+    }
+
+    private ShoppingCart getShoppingCart() {
+        User user = userService.getAuthenticatedUser();
+        return cartRepository.findByUserId(user.getId())
+                .orElseThrow(() -> new EntityNotFoundException(
+                        "Can't find shopping cart by user id: " + user.getId()));
+    }
+
+    private BigDecimal calculateTotal(Set<OrderItem> orderItems) {
+        BigDecimal total = BigDecimal.ZERO;
+        for (OrderItem orderItem : orderItems) {
+            BigDecimal itemCost = orderItem.getBook()
+                    .getPrice().multiply(BigDecimal.valueOf(orderItem.getQuantity()));
+            total = total.add(itemCost);
+        }
+        return total;
+    }
+
+    private Order saveOrderWithItems(Set<CartItem> cartItems, Order order) {
+        Set<OrderItem> orderItems = cartItems.stream()
+                .map(orderItemMapper::cartItemToOrderItem)
+                .collect(Collectors.toSet());
+        order.setOrderItems(orderItems);
+        orderRepository.save(order);
+        orderItems.forEach(orderItem -> orderItem.setOrder(order));
+        orderItemRepository.saveAll(orderItems);
```

Comment:
Is it possible to use cascade to save orderItem together with order by one statement?